src/viam/sdk/module/service.cpp

```diff
@@ -116,12 +116,10 @@ struct ModuleService::ServiceImpl : viam::module::v1::ModuleService::Service {
                                 "unable to reconfigure resource " + cfg.resource_name().name() +
                                     " as it doesn't exist.");
         }
-        try {
-            Reconfigurable::reconfigure_if_reconfigurable(res, deps, cfg);
-            res->set_log_level(cfg.get_log_level());
+
+        if (auto reconfigurable = std::dynamic_pointer_cast<Reconfigurable>(res)) {
+            reconfigurable->reconfigure(deps, cfg);
```

Comment:
⬇️

- If we aren't going to address the issue where we construct a new resource before we have destroyed the old in this PR, let's ticket that work and add a TODO down around line 132 now. I think it is something of a priority.
- Do we have any idea why the compiler wasn't able to flag everything after the block dealing with reconfigure as dead code? Actually, nevermind. I see it. If the code managed to throw something other than a std::exception, then it wasn't dead. Oops.